CentralHub.Api/Controllers/MeasurementController.cs

```diff
@@ -15,21 +15,40 @@ public sealed class MeasurementController : ControllerBase
 
     private readonly IMeasurementRepository _aggregatorRepository;
 
+    private readonly ITrackerRepository _trackerRepository;
+
     public MeasurementController(
         ILogger<MeasurementController> logger,
-        IMeasurementRepository aggregatorRepository)
+        IMeasurementRepository aggregatorRepository,
+        ITrackerRepository trackerRepository)
     {
         _logger = logger;
         _aggregatorRepository = aggregatorRepository;
+        _trackerRepository = trackerRepository;
     }
 
     [HttpPost("add")]
-    public async Task AddMeasurements(AddMeasurementsRequest addMeasurementsRequest, CancellationToken token)
+    public async Task<AddMeasurementResponse> AddMeasurements(AddMeasurementsRequest addMeasurementsRequest, CancellationToken token)
     {
-        await _aggregatorRepository.AddMeasurementsAsync(
-            addMeasurementsRequest.TrackerId,
-            addMeasurementsRequest.Measurements,
-            token);
+        var registeredTrackers = await _trackerRepository.GetRegisteredTrackers(token);
+        var unregisteredTrackers = await _trackerRepository.GetUnregisteredTrackers(token);
+
+        var tracker = registeredTrackers.SingleOrDefault(t => t.TrackerDtoId == addMeasurementsRequest.TrackerId);
+
+        if (tracker != null)
```

Comment:
Invert this if and return AddMeasurementResponse.CreateUnsuccessful() to reduce nesting